Broken Package: Boost::math

I have an issue using Boost::math and I've minimized it pretty well.

Given a CMakeLists.txt of:

```cmake
cmake_minimum_required(VERSION 3.6)

include("cmake/HunterGate.cmake")
HunterGate(
    URL "https://github.com/ruslo/hunter/archive/v0.19.126.tar.gz"
    SHA1 "78fb48ee9e636adcadbc61a6895a338733388dc4"
)

project(tmp_test)

hunter_add_package(Boost COMPONENTS math)
find_package(Boost CONFIG REQUIRED math)
```

And running in a directory with only that above CMakeLists.txt file (and huntergate of course):

```zsh
╰─➤  rm -r ~/.hunter; cmake3 .
```

Returns [CMakeOutput.log](https://overminddl1.com/tmp/CMakeOutput.log) and the [stdout/stderr](https://overminddl1.com/tmp/log.txt).
Of which the most interesting parts (at least to me) are:

```zsh
[ 87%] Performing install step for 'Boost-math'
Performing configuration checks

    - 32-bit                   : no  (cached)
    - 64-bit                   : yes (cached)
    - arm                      : no  (cached)
    - mips1                    : no  (cached)
    - power                    : no  (cached)
    - sparc                    : no  (cached)
    - x86                      : yes (cached)
    - symlinks supported       : yes (cached)
    - gcc visibility           : yes (cached)
    - long double support      : yes (cached)

Component configuration:

    - atomic                   : not building
    - chrono                   : not building
    - container                : not building
    - context                  : not building
    - coroutine                : not building
    - date_time                : not building
    - exception                : not building
    - fiber                    : not building
    - filesystem               : not building
    - graph                    : not building
    - graph_parallel           : not building
    - iostreams                : not building
    - locale                   : not building
    - log                      : not building
    - math                     : building
    - metaparse                : not building
    - mpi                      : not building
    - program_options          : not building
    - python                   : not building
    - random                   : not building
    - regex                    : not building
    - serialization            : not building
    - signals                  : not building
    - stacktrace               : not building
    - system                   : not building
    - test                     : not building
    - thread                   : not building
    - timer                    : not building
    - type_erasure             : not building
    - wave                     : not building

[100%] Completed 'Boost-math'
[100%] Built target Boost-math
-- [hunter] Build step successful (dir: /home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Build/Boost/__math)
-- [hunter] Cache saved: /home/my_user/.hunter/_Base/Cache/raw/60f6432f9483b19605d5719834db3044f02f4b40.tar.bz2
CMake Error at /home/my_user/.hunter/_Base/Download/Hunter/0.19.126/78fb48e/Unpacked/cmake/find/FindBoost.cmake:1247 (message):
  Unable to find the requested Boost libraries.

  Boost version: 1.65.1

  Boost include path:
  /home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/include

  Could not find the following static Boost libraries:

          boost_math

  No Boost libraries were found.  You may need to set BOOST_LIBRARYDIR to the
  directory containing Boost libraries or BOOST_ROOT to the location of
  Boost.
Call Stack (most recent call first):
  /home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/cmake/Boost/BoostConfig.cmake:40 (find_package)
  CMakeLists.txt:12 (find_package)



[hunter ** INTERNAL **] Library not exists: Boost_MATH_LIBRARY_DEBUG-NOTFOUND
[hunter ** INTERNAL **] [Directory:/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/cmake/Boost]

------------------------------ WIKI -------------------------------
    https://github.com/ruslo/hunter/wiki/error.internal
-------------------------------------------------------------------

CMake Error at /home/my_user/.hunter/_Base/Download/Hunter/0.19.126/78fb48e/Unpacked/cmake/modules/hunter_wiki.cmake:12 (message):
Call Stack (most recent call first):
  /home/my_user/.hunter/_Base/Download/Hunter/0.19.126/78fb48e/Unpacked/cmake/modules/hunter_internal_error.cmake:13 (hunter_wiki)
  /home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/cmake/Boost/BoostConfig.cmake:88 (hunter_internal_error)
  CMakeLists.txt:12 (find_package)


CMake Error at CMakeLists.txt:12 (find_package):
  Found package configuration file:

    /home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/cmake/Boost/BoostConfig.cmake

  but it set Boost_FOUND to FALSE so package "Boost" is considered to be NOT
  FOUND.


-- Configuring incomplete, errors occurred!
```

And yet:

```zsh
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_c99-mt-d.a
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_c99-mt.a
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_c99f-mt-d.a
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_c99f-mt.a
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_c99l-mt-d.a
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_c99l-mt.a
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_tr1-mt-d.a
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_tr1-mt.a
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_tr1f-mt-d.a
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_tr1f-mt.a
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_tr1l-mt-d.a
/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/libboost_math_tr1l-mt.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_tr1-mt.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_tr1f-mt.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_tr1l-mt.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_c99-mt.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_c99f-mt.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_c99l-mt.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_tr1-mt-d.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_tr1f-mt-d.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_tr1l-mt-d.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_c99-mt-d.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_c99f-mt-d.a
/home/my_user/.hunter/_Base/Cellar/60f6432f9483b19605d5719834db3044f02f4b40/60f6432/raw/lib/libboost_math_c99l-mt-d.a
```
And the headers and all are there as expected, including the Config file.  And as you can see in the actual log files, it did all compile and install properly.

Running the same command without removing `~/.hunter` just prints out this in a fresh directory with just the CMakeLists.txt file (and huntergate of course):

```zsh
─➤  rm -r _3rdParty CMakeFiles CMakeCache.txt; cmake3 .                                                                        1 ↵
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- [hunter] Calculating Toolchain-SHA1
-- [hunter] Calculating Config-SHA1
-- [hunter] HUNTER_ROOT: /home/my_user/.hunter
-- [hunter] [ Hunter-ID: 78fb48e | Toolchain-ID: a397f94 | Config-ID: 9d1abe4 ]
-- [hunter] BOOST_ROOT: /home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install (ver.: 1.65.1)
-- [hunter] BOOST_ROOT: /home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install (ver.: 1.65.1)
CMake Error at /home/my_user/.hunter/_Base/Download/Hunter/0.19.126/78fb48e/Unpacked/cmake/find/FindBoost.cmake:1247 (message):
  Unable to find the requested Boost libraries.

  Boost version: 1.65.1

  Boost include path:
  /home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/include

  Could not find the following static Boost libraries:

          boost_math

  No Boost libraries were found.  You may need to set BOOST_LIBRARYDIR to the
  directory containing Boost libraries or BOOST_ROOT to the location of
  Boost.
Call Stack (most recent call first):
  /home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/cmake/Boost/BoostConfig.cmake:40 (find_package)
  CMakeLists.txt:12 (find_package)



[hunter ** INTERNAL **] Library not exists: Boost_MATH_LIBRARY_DEBUG-NOTFOUND
[hunter ** INTERNAL **] [Directory:/home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/cmake/Boost]

------------------------------ WIKI -------------------------------
    https://github.com/ruslo/hunter/wiki/error.internal
-------------------------------------------------------------------

CMake Error at /home/my_user/.hunter/_Base/Download/Hunter/0.19.126/78fb48e/Unpacked/cmake/modules/hunter_wiki.cmake:12 (message):
Call Stack (most recent call first):
  /home/my_user/.hunter/_Base/Download/Hunter/0.19.126/78fb48e/Unpacked/cmake/modules/hunter_internal_error.cmake:13 (hunter_wiki)
  /home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/cmake/Boost/BoostConfig.cmake:88 (hunter_internal_error)
  CMakeLists.txt:12 (find_package)


CMake Error at CMakeLists.txt:12 (find_package):
  Found package configuration file:

    /home/my_user/.hunter/_Base/78fb48e/a397f94/9d1abe4/Install/lib/cmake/Boost/BoostConfig.cmake

  but it set Boost_FOUND to FALSE so package "Boost" is considered to be NOT
  FOUND.


-- Configuring incomplete, errors occurred!
See also "/home/my_user/tmp/test_projects/tmp/CMakeFiles/CMakeOutput.log".
```

My system is:

```zsh
─➤  uname -a
Linux servername 3.10.0-514.26.2.el7.x86_64 #1 SMP Fri Jun 30 05:26:04 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```

--- a/docs/packages/pkg/Boost.rst
+++ b/docs/packages/pkg/Boost.rst
@@ -44,12 +44,12 @@ List of components (other libraries are header-only):
 - ``locale``
 - ``log``
 - ``math`` - Requires Special names for ``find_package`` and ``target_link_libraries``, choose one of:
-  - ``math_c99``
-  - ``math_c99f``
-  - ``math_c99l``
-  - ``math_tr1``
-  - ``math_tr1f``
-  - ``math_tr1l``
+   - ``math_c99``
+   - ``math_c99f``
+   - ``math_c99l``
+   - ``math_tr1``
+   - ``math_tr1f``
+   - ``math_tr1l``
 - ``mpi``
 - ``program_options``
 - ``python``

--- a/docs/packages/pkg/Boost.rst
+++ b/docs/packages/pkg/Boost.rst
@@ -27,6 +27,7 @@ Examples:
 - `Boost-system <https://github.com/ruslo/hunter/blob/master/examples/Boost-system/CMakeLists.txt>`__
 - `Boost-iostreams <https://github.com/ruslo/hunter/blob/master/examples/Boost-iostreams/CMakeLists.txt>`__
 - `Boost-filesystem <https://github.com/ruslo/hunter/blob/master/examples/Boost-filesystem/CMakeLists.txt>`__
+- `Boost-math <https://github.com/ruslo/hunter/blob/master/examples/Boost-math/CMakeLists.txt>`__
 
 List of components (other libraries are header-only):
 
@@ -42,7 +43,13 @@ List of components (other libraries are header-only):
 - ``iostreams``
 - ``locale``
 - ``log``
-- ``math``
+- ``math`` - Requires Special names for ``find_package`` and ``target_link_libraries``, choose one of:
+  - ``math_c99``
+  - ``math_c99f``
+  - ``math_c99l``
+  - ``math_tr1``
+  - ``math_tr1f``
+  - ``math_tr1l``
 - ``mpi``
 - ``program_options``
 - ``python``

--- a/docs/packages/pkg/Boost.rst
+++ b/docs/packages/pkg/Boost.rst
@@ -90,17 +90,16 @@ For example:
 -  `boost.iostreams
    options <http://www.boost.org/doc/libs/1_57_0/libs/iostreams/doc/index.html?path=7>`__
 
-Boost::Math
+Math
 -----------
 
 When using Boost Math you will need to link in the libraries, however these are not named ``math`` but
-rather are individual based on what you need to link it, the easist of which is to link in all parts:
+rather are individual based on what you need to link it, the easiest of which is to link in all parts:
 
 .. code-block:: cmake
 
     hunter_add_package(Boost COMPONENTS math)
     find_package(Boost CONFIG REQUIRED math_c99 math_c99f math_c99l math_tr1 math_tr1f math_tr1l)
-    include_directories(${Boost_INCLUDE_DIRS})
     target_link_libraries(...
       Boost::math_c99
       Boost::math_c99f
@@ -110,23 +109,12 @@ rather are individual based on what you need to link it, the easist of which is 
       Boost::math_tr1l
     )
 
-Or implicitly:
-
-.. code-block:: cmake
-
-    hunter_add_package(Boost COMPONENTS math)
-    find_package(Boost CONFIG REQUIRED math_c99 math_c99f math_c99l math_tr1 math_tr1f math_tr1l)
-    include_directories(${Boost_INCLUDE_DIRS})
-    target_link_libraries(... ${Boost_LIBRARIES})
-
 If you are using only the header-only parts of Boost::Math then the libraries can be ignored:
 
 .. code-block:: cmake
 
     hunter_add_package(Boost COMPONENTS math)
     find_package(Boost CONFIG REQUIRED)
-    include_directories(${Boost_INCLUDE_DIRS})
-    target_link_libraries(... ${Boost_LIBRARIES})
 
 Bugs
 ----

--- a/docs/packages/pkg/Boost.rst
+++ b/docs/packages/pkg/Boost.rst
@@ -43,13 +43,7 @@ List of components (other libraries are header-only):
 - ``iostreams``
 - ``locale``
 - ``log``
-- ``math`` - Requires Special names for ``find_package`` and ``target_link_libraries``, choose one of:
-   - ``math_c99``
-   - ``math_c99f``
-   - ``math_c99l``
-   - ``math_tr1``
-   - ``math_tr1f``
-   - ``math_tr1l``
+- ``math``
 - ``mpi``
 - ``program_options``
 - ``python``
@@ -95,6 +89,44 @@ For example:
 
 -  `boost.iostreams
    options <http://www.boost.org/doc/libs/1_57_0/libs/iostreams/doc/index.html?path=7>`__
+
+Boost::Math
+-----------
+
+When using Boost Math you will need to link in the libraries, however these are not named ``math`` but
+rather are individual based on what you need to link it, the easist of which is to link in all parts:
+
+.. code-block:: cmake
+
+    hunter_add_package(Boost COMPONENTS math)
+    find_package(Boost CONFIG REQUIRED math_c99 math_c99f math_c99l math_tr1 math_tr1f math_tr1l)
+    include_directories(${Boost_INCLUDE_DIRS})
+    target_link_libraries(...
+      Boost::math_c99
+      Boost::math_c99f
+      Boost::math_c99l
+      Boost::math_tr1
+      Boost::math_tr1f
+      Boost::math_tr1l
+    )
+
+Or implicitly:
+
+.. code-block:: cmake
+
+    hunter_add_package(Boost COMPONENTS math)
+    find_package(Boost CONFIG REQUIRED math_c99 math_c99f math_c99l math_tr1 math_tr1f math_tr1l)
+    include_directories(${Boost_INCLUDE_DIRS})
+    target_link_libraries(... ${Boost_LIBRARIES})
+
+If you are using only the header-only parts of Boost::Math then the libraries can be ignored:
+
+.. code-block:: cmake
+
+    hunter_add_package(Boost COMPONENTS math)
+    find_package(Boost CONFIG REQUIRED)
+    include_directories(${Boost_INCLUDE_DIRS})
+    target_link_libraries(... ${Boost_LIBRARIES})
 
 Bugs
 ----

--- a/examples/Boost-math/CMakeLists.txt
+++ b/examples/Boost-math/CMakeLists.txt
@@ -12,29 +12,17 @@ project(download-boost)
 # download boost
 hunter_add_package(Boost COMPONENTS math)
 
-# now boost can be used
-# you need to choose which form of Boost::math you want
-#find_package(Boost CONFIG REQUIRED math_c99)
-#find_package(Boost CONFIG REQUIRED math_c99f)
-#find_package(Boost CONFIG REQUIRED math_c99l)
-find_package(Boost CONFIG REQUIRED math_tr1)
-#find_package(Boost CONFIG REQUIRED math_tr1f)
-#find_package(Boost CONFIG REQUIRED math_tr1l)
+# now boost::math can be used
+# link in all the parts of boost math, or leave out unneeded parts
+find_package(Boost CONFIG REQUIRED math_c99 math_c99f math_c99l math_tr1 math_tr1f math_tr1l)
 add_executable(foo foo.cpp)
-target_link_libraries(foo Boost::math_tr1)
-
-# Compatibility mode
-set(Boost_USE_STATIC_LIBS ON)
-if(MSVC)
-  add_definitions(-DBOOST_ALL_NO_LIB=1)
-endif()
-# you need to choose which form of Boost::math you want
-#find_package(Boost CONFIG REQUIRED math_c99)
-#find_package(Boost CONFIG REQUIRED math_c99f)
-#find_package(Boost CONFIG REQUIRED math_c99l)
-find_package(Boost CONFIG REQUIRED math_tr1)
-#find_package(Boost CONFIG REQUIRED math_tr1f)
-#find_package(Boost CONFIG REQUIRED math_tr1l)
-include_directories(${Boost_INCLUDE_DIRS})
-target_link_libraries(foo ${Boost_LIBRARIES})
+# link in the libraries as well for whatever is found above in find_package
+target_link_libraries(foo
+  Boost::math_c99
+  Boost::math_c99f
+  Boost::math_c99l
+  Boost::math_tr1
+  Boost::math_tr1f
+  Boost::math_tr1l
+)
 

--- a/examples/Boost-math/CMakeLists.txt
+++ b/examples/Boost-math/CMakeLists.txt
@@ -12,7 +12,7 @@ project(download-boost)
 # download boost
 hunter_add_package(Boost COMPONENTS math)
 
-# now boost::math can be used
+# now boost::math_* targets
 # link in all the parts of boost math, or leave out unneeded parts
 find_package(Boost CONFIG REQUIRED math_c99 math_c99f math_c99l math_tr1 math_tr1f math_tr1l)
 add_executable(foo foo.cpp)

--- a/examples/Boost-math/CMakeLists.txt
+++ b/examples/Boost-math/CMakeLists.txt
@@ -1,0 +1,40 @@
+# Copyright (c) 2013, Ruslan Baratov
+# All rights reserved.
+
+cmake_minimum_required(VERSION 3.0)
+
+# Emulate HunterGate:
+# * https://github.com/hunter-packages/gate
+include("../common.cmake")
+
+project(download-boost)
+
+# download boost
+hunter_add_package(Boost COMPONENTS math)
+
+# now boost can be used
+# you need to choose which form of Boost::math you want
+#find_package(Boost CONFIG REQUIRED math_c99)
+#find_package(Boost CONFIG REQUIRED math_c99f)
+#find_package(Boost CONFIG REQUIRED math_c99l)
+find_package(Boost CONFIG REQUIRED math_tr1)
+#find_package(Boost CONFIG REQUIRED math_tr1f)
+#find_package(Boost CONFIG REQUIRED math_tr1l)
+add_executable(foo foo.cpp)
+target_link_libraries(foo Boost::math_tr1)
+
+# Compatibility mode
+set(Boost_USE_STATIC_LIBS ON)
+if(MSVC)
+  add_definitions(-DBOOST_ALL_NO_LIB=1)
+endif()
+# you need to choose which form of Boost::math you want
+#find_package(Boost CONFIG REQUIRED math_c99)
+#find_package(Boost CONFIG REQUIRED math_c99f)
+#find_package(Boost CONFIG REQUIRED math_c99l)
+find_package(Boost CONFIG REQUIRED math_tr1)
+#find_package(Boost CONFIG REQUIRED math_tr1f)
+#find_package(Boost CONFIG REQUIRED math_tr1l)
+include_directories(${Boost_INCLUDE_DIRS})
+target_link_libraries(foo ${Boost_LIBRARIES})
+

--- a/examples/Boost-math/foo.cpp
+++ b/examples/Boost-math/foo.cpp
@@ -1,0 +1,5 @@
+#include <boost/math/tr1.hpp>
+
+int main() {
+}
+

